packages/table/src/TableBody.tsx

```diff
@@ -1,17 +1,58 @@
-import React from 'react';
-import { useTableContext } from './TableContext';
+import React, { useMemo, useCallback } from 'react';
+import { getDataComparisonFunction, useTableContext } from './TableContext';
 import { TableProps } from './Table';
 
 type TableBodyProps<Shape> = Pick<TableProps<Shape>, 'children'>;
 
+/**
+ * Hook that avoids re-rendering children when the data used to
+ * render them hasn't changed, even if their ordering has changed.
+ */
+function useRenderedChildren<Datum>(
+  data: Array<Datum>,
+  renderFunction: React.FunctionComponent<Datum>,
+  compareFn?: (a: Datum, b: Datum) => number,
+): Array<React.ReactNode> {
+  const resultMap = useMemo(() => {
+    const resultMap = new Map<Datum, React.ReactNode>();
+    data.forEach((datum, index) =>
+      resultMap.set(
+        datum,
+        <React.Fragment key={index}>{renderFunction(datum)}</React.Fragment>,
```

Comment:
We don't sadly have anything more stable than `index` to use as the key, right? Cuz that would gain us some render optimizations?